src/server/game/Entities/Object/Object.cpp

```diff
@@ -997,6 +997,80 @@ void MovementInfo::OutDebug()
         TC_LOG_DEBUG("misc", "splineElevation: %f", splineElevation);
 }
 
+void MovementInfo::WriteContentIntoPacket(ByteBuffer * data, bool includeGuid /* = false*/) const
```

Comment:
If you are refactoring this then why are you creating a new function/name for this instead of unifying packet building just like we had done in master branch? https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Server/Packets/MovementPackets.cpp#L25 (obviously maintain 3.3.5 structure)

The base for packet classes is already present in 3.3.5, see https://github.com/TrinityCore/TrinityCore/tree/3.3.5/src/server/game/Server/Packets